src/main/java/org/zalando/nakadi/domain/NakadiCursor.java

```diff
@@ -1,21 +1,43 @@
 package org.zalando.nakadi.domain;
 
+import com.google.common.base.Preconditions;
 import java.util.Objects;
 
 public class NakadiCursor {
-    private final String topic;
+    public static final int VERSION_LENGTH = 3;
+    public enum Version {
+        ZERO("000"),
+        ONE("C00"),;
+        public final String code;
+
+        Version(final String code) {
+            Preconditions.checkArgument(
+                    code.length() == VERSION_LENGTH,
+                    "Version field length should be equal to " + VERSION_LENGTH);
+            this.code = code;
+        }
+    }
+
+    private final Timeline timeline;
     private final String partition;
     // NO BEGIN OR END HERE!
```

Comment:
We actually don't have any END for offset. I think it's better to remove it from this comment not to confuse people.